### ISSUE TITLE
Ryujinx Vulkan fullscreen on Steam Deck has black screen on all games

### ISSUE THREAD

**Issue by Pixxel123** [Ryujinx/Ryujinx#3874]
**Note:** This does not seem to happen on Steam OS Stable, with either Stable or Beta Steam Client. Using master 1.1.389.

When playing any game on Ryujinx on the Steam Deck, the screen goes black and is only brought back by using touch input. Audio and the game still continue running while "behind" this black screen.

There have also been reports that this does not happen in windowed mode.

Using latest master 1.1.365.

I have attached a video of the issue (showing that audio still plays and the game still runs), as well as a log file.

https://user-images.githubusercontent.com/36304206/202900318-1e027950-674c-4cc1-9f64-118dab691f83.mp4

[Ryujinx_1.1.365_2022-11-20_11-36-44.log](https://github.com/Ryujinx/Ryujinx/files/10049535/Ryujinx_1.1.365_2022-11-20_11-36-44.log)

**Comment by Serfrost** [comment]
For further context, this problem is likely linked to Issues #1850 and #2080

**Comment by Kurosawa79** [comment]
Have the same problem

**Comment by TSRBerry** [comment]
> For further context, this problem is likely linked to the issue mentioned in PRs #1850 and #2080

I highly doubt these are related to this issue.

**Comment by MadCabbit** [comment]
This also happens on OpenGL and I believe its only when running the Steam Deck preview branch.  Workaround is to not run fullscreen or revert to the stable branch of SteamOS (though I haven't tested that one myself, going by this reddit post: https://www.reddit.com/r/SteamDeck/comments/yva7sa/emudeck_ryujinx_fullscreen_issue_in_gaming_mode/ )

**Comment by Serfrost** [comment]
> I highly doubt these are related to this issue.
My bad, they just seemed rather similar. Figured it wouldn't hurt to reference in case they were.

**Comment by swangy** [comment]
i changed to stable and still
happening :(

**Comment by TSRBerry** [comment]
Change from stable to beta, install all the updates, then change to preview install all the updates and change back to stable and install all the updates.

For some unknown reason this worked for 2 people already to fix this issue and we have no clue why or what's causing this.

**Comment by sg-qwt** [comment]
Same issue running in steam deck full-screen gamescope mode, and it does not happen in windowed mode. Note my deck was always on stable branch, never changed to beta or anything else.

**Comment by Dragonisser** [comment]
Same, latest stable and ryujinx but it just turns off. Only solution is running it with `--boredered`

**Comment by TSRBerry** [comment]
@sg-qwt @Dragonisser Could guys try the "solution" above and report back if it works for you too?

https://github.com/Ryujinx/Ryujinx/issues/3874#issuecomment-1345385910

**Comment by swangy** [comment]
@TSRBerry tried that rotation through all builds, updating in between.  no workie still

**Comment by TSRBerry** [comment]
Ah too bad.

We believe that this is more of a Valve issue than a Ryujinx one, especially since this weird solution works for some people and we aren't able to reproduce the issue.

No clue what's going on and why this happens.

**Comment by swangy** [comment]
strangely, doesnt happen with yuzu emulation?

**Comment by TSRBerry** [comment]
How is that related to this issue?

**Comment by swangy** [comment]
it appears to be something implementation specific to ryujinx - when the cursor is moved by touching the screen, the game shows up, and the whole screen goes blank after a timeout interval when cursor is hidden.

**Comment by Dragonisser** [comment]
> it appears to be something implementation specific to ryujinx - when the cursor is moved by touching the screen, the game shows up, and the whole screen goes blank after a timeout interval when cursor is hidden.

What happens if you turn off auto hiding the cursor?

**Comment by MadCabbit** [comment]
I can't speak for stable, but on the latest preview and latest version of Ryujinx, the problem appears to have gone away and 
is working in fullscreen for me.

**Comment by swangy** [comment]
> What happens if you turn off auto hiding the cursor?

tried unchecking that box, cursor still hides and screen goes blanks. gonna try updating to today's Preview build per @MadCabbit

**Comment by EmulationFanatic** [comment]
> I can't speak for stable, but on the latest preview and latest version of Ryujinx, the problem appears to have gone away and is working in fullscreen for me.

What is "the latest preview" version of Ryujinx?

**Comment by Dragonisser** [comment]
> > I can't speak for stable, but on the latest preview and latest version of Ryujinx, the problem appears to have gone away and is working in fullscreen for me.
> 
> What is "the latest preview" version of Ryujinx?

latest pewview of steamos (steamdeck) and latest stable of ryujinx

**Comment by MadCabbit** [comment]
> > > I can't speak for stable, but on the latest preview and latest version of Ryujinx, the problem appears to have gone away and is working in fullscreen for me.
> > 
> > 
> > What is "the latest preview" version of Ryujinx?
> 
> latest preview of steamos (steamdeck) and latest stable of ryujinx

Yes, that.

**Comment by umurpza** [comment]
Someone confirm so I can switch :)

**Comment by Dragonisser** [comment]
> I can't speak for stable, but on the latest preview and latest version of Ryujinx, the problem appears to have gone away and is working in fullscreen for me.

Still the same issue even on preview.

**Comment by swangy** [comment]
ya, still an issue here as well. latest updated preview and latest ryujinx

**Comment by b12e** [comment]
Same here - if it helps: it does display the game content when in the steam menu (the one where you can eg change controller input), and it also shows perfectly fine in desktop mode. You can see the game being rendered in the background.

Yuzu doesn't have these issues, but they have other problems 😄

**Comment by 0xDEADCADE** [comment]
Latest update 3.4 seems to have pushed the previously "Preview" changes causing this to Stable. Most Steam Deck users will be affected by this from now on.

Running the game in "Desktop Mode", Windowed, or with OpenGL appears to "fix" the issue. It is Ryujinx specific, as no other emulator or application has the same issue.

**Comment by lamaledetta** [comment]
I have the same issue after updating to SteamOs 3.4.2

**Comment by Mhytee** [comment]
Same issue after updating to Steam OS 3.4.1, running the game windowed has worked for me.

**Comment by MadCabbit** [comment]
I had forgotten that the problem seemed to affect me in OpenGL before.  I can confirm I still have the problem in Vulkan, but not OpenGL.

**Comment by jlar0che** [comment]
Same exact issue here. Started after updating to SteamOs 3.4.2. I too have always been on the stable branch. My Ryujinx version is 1.1.488 and, like others, Windowed mode “fixes” the issue...

Didn't realize this was so widespread and I created another issue (https://github.com/Ryujinx/Ryujinx/issues/4167).

**Comment by TSRBerry** [comment]
If there is a new stable version I'll have another try at this issue in a few days.

**Comment by thuutien** [comment]
Has same issue here as well. Any update on fixing this issue guys?

**Comment by thebang2** [comment]
A workaround is to don´t start the game in fullscreen.

**Comment by UltraMoonTrainer** [comment]
Im experiencing this issue myself as well, i have to keep moving my mouse or tap the screen to keep it on

**Comment by chrisasma103** [comment]
> Im experiencing this issue myself as well, i have to keep moving my mouse or tap the screen to keep it on

Yeah, this problem is really annoying. Like others have said, the temporary solution is to play games in windowed mode or to play using OpenGl instead of Vulkan. Hopefully, this issue will be fixed soon

**Comment by TSRBerry** [comment]
Good news, I am now able to reproduce the issue! I'll investigate and hopefully we can find a fix.

**Comment by riperiperi** [comment]
> it appears to be something implementation specific to ryujinx - when the cursor is moved by touching the screen, the game shows up, and the whole screen goes blank after a timeout interval when cursor is hidden.

How would this be specific to Ryujinx? We don't implement a virtual cursor, nor do we react to it.

**Comment by TSRBerry** [comment]
The best workaround I have for you right now is using [Avalonia](https://github.com/Ryujinx/release-channel-master/releases/download/1.1.492/test-ava-ryujinx-1.1.492-linux_x64.tar.gz) which doesn't have this issue.

At the moment I think this might be related to SDL2, since touch input also doesn't work when using Vulkan..

**Comment by jlar0che** [comment]
Running v1.1.492 and still have the issue.

**Comment by jonpol01** [comment]
Current workaround was setting to OpenGL.. fullscreen worked and gameplay is fine.

**Comment by AndehX** [comment]
> > Current workaround was setting to OpenGL.. fullscreen worked and gameplay is fine.
> 
> If you want to use Vulkan in fullscreen on the Steam Deck without issues you'll need to use Avalonia for now. I said it's the best workaround I have for you for now, because it doesn't have this or any other Steam Deck specific issues.

This doesn't work.

**Comment by TSRBerry** [comment]
@AndehX This is not helpful at all. Could you please add some details what exactly doesn't work?

I tested this workaround, so I know it works for sure.

**Comment by AndehX** [comment]
> @AndehX This is not helpful at all. Could you please add some details what exactly doesn't work?
> 
> I tested this workaround, so I know it works for sure.

I'm not entirely sure how else to explain it. All I can say is I ran the Ava version in desktop mode and the game doesn't display, the window just becomes invisible, and I'm unable to close the emulator. All I can do is manually reset my Steam deck.

**Comment by jonpol01** [comment]
> > Current workaround was setting to OpenGL.. fullscreen worked and gameplay is fine.
> 
> 
> 
> If you want to use Vulkan in fullscreen on the Steam Deck without issues you'll need to use Avalonia for now.
> 
> I said it's the best workaround I have for you for now, because it doesn't have this or any other Steam Deck specific issues.

Thanks, but at this point I dont really care either OpenGl or Vulkan.
I can play good enough, and installing Ava version is not worth the hassle. 😆🫢

**Comment by Zhaltor** [comment]
> > @AndehX This is not helpful at all. Could you please add some details what exactly doesn't work?
> > I tested this workaround, so I know it works for sure.
> 
> I'm not entirely sure how else to explain it. All I can say is I ran the Ava version in desktop mode and the game doesn't display, the window just becomes invisible, and I'm unable to close the emulator. All I can do is manually reset my Steam deck.

have the exact same issue, a game will play but no video output is made, the actual emulator window becomes transparent, happens only on the ava build of the emulator

**Comment by TSRBerry** [comment]
Are you using game mode and the stable branch of SteamOS?

**Comment by Zhaltor** [comment]
> Are you using game mode and the stable branch of SteamOS?

yes, and yes, in desktop mode the window turns transparent, in game mode the screen remains black

**Comment by TSRBerry** [comment]
This is really confusing. Could you send a video and attach a log file of this happening?
I tested this and had none of the issues you described.

**Comment by Zhaltor** [comment]
[Ryujinx_1.1.0-ldn3.0.1_2022-12-30_23-22-11.log](https://github.com/Ryujinx/Ryujinx/files/10326530/Ryujinx_1.1.0-ldn3.0.1_2022-12-30_23-22-11.log)

https://drive.google.com/file/d/1wiMMiiCt5GzPgumAYB6w2HxvW0j_7Vf9/view?usp=sharing

In Desktop Mode it completely freezes up the device, on Game mode it closes fine but does basically the same thing

**Comment by TSRBerry** [comment]
You are using LDN-3.0.1 and not latest master. Does the issue occur using that as well?

**Comment by Zhaltor** [comment]
I do apologise, should have tested that. The issue does not seem to happen on the latest Ava build, but the dropdown menus on it do not seem to work either so I can't really get controls setup

**Comment by TSRBerry** [comment]
That's a known Avalonia issue sadly. But I'm glad the latest release is working correctly and it's not another issue I can't reproduce!

**Comment by AndehX** [comment]
> You are using LDN-3.0.1 and not latest master. Does the issue occur using that as well?

forgive my stupidity, but where do you get the latest master? I don't see it anywhere?

**Comment by AgentLoneStar007** [comment]
Happening to me as well. Switching to OpenGL didn't fix it, but disabling full screen did.

**Comment by TSRBerry** [comment]
> > You are using LDN-3.0.1 and not latest master. Does the issue occur using that as well?
> 
> forgive my stupidity, but where do you get the latest master? I don't see it anywhere?

You can find the latest release here: https://github.com/Ryujinx/release-channel-master/releases
Make sure to use the test-ava build.

**Comment by AndehX** [comment]
> > > You are using LDN-3.0.1 and not latest master. Does the issue occur using that as well?
> > 
> > 
> > forgive my stupidity, but where do you get the latest master? I don't see it anywhere?
> 
> You can find the latest release here: [Ryujinx/release-channel-master/releases](https://github.com/Ryujinx/release-channel-master/releases) Make sure to use the test-ava build.

forgive my stupidity yet again, but im not seeing the LDN build anywhere?

**Comment by TSRBerry** [comment]
> > > > You are using LDN-3.0.1 and not latest master. Does the issue occur using that as well?
> > > 
> > > 
> > > forgive my stupidity, but where do you get the latest master? I don't see it anywhere?
> > 
> > You can find the latest release here: [Ryujinx/release-channel-master/releases](https://github.com/Ryujinx/release-channel-master/releases) Make sure to use the test-ava build.
> 
> forgive my stupidity yet again, but im not seeing the LDN build anywhere?
> 
> Also, the latest ava build does indeed fix the issue.  The emulator now behaves normally on Steam Deck.

I don't think we fixed all the Avalonia issues for LDN 3.0.1 already, so you'd need to wait for a new LDN release to also get that working :/

**Comment by thuutien** [comment]
Updated to the latest version. Issue is still there

**Comment by formicsapien** [comment]
is there an ETA on when I'll be able to use ryujinx on my steam deck in fullscreen w/ vulkan again?  This has been an issue for months.

**Comment by TSRBerry** [comment]
You can. Just use Avalonia as I explained above.

**Comment by formicsapien** [comment]
> You can. Just use Avalonia as I explained above.

Hey, yeah thanks but I wasn't wondering how I can use a workaround, I was wondering when there will be a fix so the 99.99% of users, the ones who don't come to this issue thread, will have a fix.

**Comment by riperiperi** [comment]
Probably when Valve fixes it. It’s not up to us to look into weird quirks steamos has with perfectly normal windowing behaviours.

**Comment by TSRBerry** [comment]
Although we did look into it several times, we couldn't figure it out and have no idea how this could be an issue on our side.

**Comment by EmulationFanatic** [comment]
> > You can. Just use Avalonia as I explained above.
> 
> Hey, yeah thanks but I wasn't wondering how I can use a workaround, I was wondering when there will be a fix so the 99.99% of users, the ones who don't come to this issue thread, will have a fix. Or am I wrongfully assuming this can be fixed??
> 
> EDIT: I have no idea what Avalonia is nor how to use it. Will it be something the main branch uses soon? Would that be the "fix"?

If you had read the recent comments on this issue thread you'd have seen a link.

**Comment by lamaledetta** [comment]
Could you add an Avalonia installation guide to this thread please? Would be very useful to new users. Thanks

**Comment by MadCabbit** [comment]
> Could you add an Avalonia installation guide to this thread please? Would be very useful to new users. Thanks

Since this is is a Steam Deck related issue and most people use Ryujinx through something like EmuDeck, it would probably be more productive to have this discussion somewhere else like the EmuDeck discord about pulling in Avalonia as an option.  From the perspective of a project on Github, you'd download and unpack the tarball that's been pointed out in the same way you'd unpack the standard release tarball, so there's not much more to say from the point of view of the developers here.

**Comment by lamaledetta** [comment]
Come on guys, don’t be so hard on us. We only want keep playing with Ryujinx because we love it. 
Just tell us where to put this publish dir please.

**Comment by MadCabbit** [comment]
Due to the Steam Deck having a read-only filesystem, EmuDeck uses either AppImages or Flatpak for emulator installs.  There is a version of Ryujinx on Flathub, but I don't know of any Flatpak or AppImage version of Avalonia, so that would probably need to come first.  Trying to use the standard tarball on a SD would be messier.

**Comment by lamaledetta** [comment]
OK I have managed to make it work, just put publish dir from AVA tarball into /home/deck/Applications (make a backup of old publish dir before). 

Everything works but audio. Audio is really crappy and distorted. Any idea?

**Comment by Dragonisser** [comment]
> Come on guys, don’t be so hard on us. We only want keep playing with Ryujinx because we love it. Just tell us where to put this publish dir please.

You can literally just download the linked test-ava-rujinx.tar.gz unwrap and it launch it.

**Comment by thuutien** [comment]
Update and try guys

**Comment by lamaledetta** [comment]
No luck on my side. Problem still there.

**Comment by Dragonisser** [comment]
> Update and try guys

For me its now just freezes while building shaders and then the app dies.

**Comment by jflatt** [comment]
Works well for me now, on 1.1.540

**Comment by TSRBerry** [comment]
There were no changes made on our side that could address this issue.

The recommended workaround is still using Avalonia.

**Comment by shineofleo** [comment]
I found the same issue and I am glad I got this post. I made a full investigation and I think perfect workaround is possible.

Basically I installed two versions of ryujinx, one from flatpak, which from the Discver store, and one from Emudeck, as below

- 1.1.557 - from flatpak, configure path ~/.var/app/org.ryujinx.Ryujinx/config/Ryujinx/, and it need use "flatpak run xxxx" format to run.
- 1.1.565 - from ryujinx website and install. configure path ~/.config/Ryujinx/ and the executable is in ~/Applications/publish/

I tested in two environments: steamos desktop mode and game mode. in desktop mode, just run the executable (flatpak or directly), for game mode, I use emudeck - to be more exactly, use steam rom manager to call the executable plus the rom path, or extra arguments.

NOTE: flatpak need permission to access the rom file so "--filesystem=home" or similar is essential or ryujinx will report invalid application error.

**What I can say is that, in DESKTOP mode, both versions run fine, without black screen issue, however, I had difficult time to enable the controller (another story). In GAME mode, the standalone (1.1.565 here) version gets black screen issue, while flatpak version (1.1.557) runs fine!**

I made all kinds of comparison, but got nothing. I don't think it's due to the version. I guess it is the steamos system limitation or similar. Since the program runs well just not foreground, and any mouse movements would call the graph up, for a few seconds... very annoying yes. Maybe permission? flatpak owned by root perhaps it has higher privilege? Too tired to chown all the files... So I stopped here.

From above, I saw ava-ryujinx.tar.gz, so I gave it a try. Actually, in the standalone version, there is a Rynjinx.sh which contains three versions of latest release:

- Ryujinx - the common one
- Ryujinx.Ava - the new Avalonia GUI version
- Ryujinx.Headless.SDL2 - the headless version

I tried Avalonia version, which made the black screen gone! However my mouse coujld not work properly in it (this is in game mode, desktop mode is fine). Then I told my self "why not use headless?" because in game mode, all I need is a commandline to run it, and there is no need to do any modification - which should be done beforehands.

And it worked. In the end, after the steam rom manager set the command right, a simple click in game mode would bring game alive, and everything seems fine!

There is not much information about headless mode, so a very short note to make the game run:

- Extract the release package, and run ./Ryujinx.sh. (this shell script includes all three version and pick the one available, which is better especially for future scripting, such as emuDeck launcher), it will display help information.
- Rom path is mandatory here. but steam rom manager will take care of it.
- Adding "--root-date-dir=XXX" to point to the config path, including keys, and all the other configurations. Just use ~/.config/Ryujinx/ .
- Adding a controller for player1 by "--input-id-1=XXX", however, you may have no idea what steam controller id is, use this:

`./Ryujinx.sh --list-inputs-ids="true"`

it will return a list of avaliable input ids, mine is:

`00:00:00.007 |I| Application Load: Input Ids:
00:00:00.015 |I| Application Load: - 0 ("All keyboards")
00:00:00.031 |I| Application Load: - 0-17f60003-28de-0000-0512-000010010000 ("Steam Deck")
`
Here the steam controller id is 0-17f60003-28de-0000-0512-000010010000.

- So now, the final command line is:

`./Ryujinx.sh --root-data-dir="/home/deck/.config/Ryujinx/" --input-id-1="YOUR CONTROLLER ID" "YOUR ROM PATH"`

In steam rom manager, just add '--root-data-dir="/home/deck/.config/Ryujinx/" --input-id-1="YOUR CONTROLLER ID"' in the argument field. If there is any other adjustment, add more arguments in headless commandline, and/or add to the config file. For example, create new input profile, and definte mapping in json, then load it. 

This should bring a nice full screen ryujinx running in steamos game mode.

**Comment by TSRBerry** [comment]
Could you explain the mouse issue you experienced with Avalonia? As I said before I expect Avalonia to work without issues on the Steam Deck and it's the recommended way to use Ryujinx there.
Headless had the same blackscreen issue in my tests as Gtk (the main GUI), does Headless not have this issue for you?

**Comment by shineofleo** [comment]
> Could you explain the mouse issue you experienced with Avalonia? As I said before I expect Avalonia to work without issues on the Steam Deck and it's the recommended way to use Ryujinx there. Headless had the same blackscreen issue in my tests as Gtk (the main GUI), does Headless not have this issue for you?

Thanks. As the mouse issue - I launched steam deck into game mode, and run Avalonia version of the ryujinx. I connected keyboard and mouse, called the menu (by F4 I think). I could see the cursor and pointed it to menu item and clicked, there was no response. However I can see the item was 'pressed'.

Headless works in my deck game mode. I loaded it with steam rom manager.

I find the game mode and desktop mode in steam deck vastly differ, speaking of controller on the deck, and screen such as the black screen issue in this post. Individual program runs fine, just the extra 'layer' causes quite a few trouble. I read some where that the game mode loads something extra, such as the controller driver. Maybe that explains the screen behavor as well.

Sorry if above description is not accurate but I am willing to contribute. Anything I can help I will be glad to assist.

**Comment by TSRBerry** [comment]
The black screen issue is probably related to `gamescope` the window manager which is used in gaming mode, but it could also be be the SteamUI interfering with applications in gaming mode. (Maybe something related to focus or window state, who knows)

> Headless works in my deck game mode. I loaded it with steam rom manager.

Oh that's interesting, I'll need to test that again when I have the time.

> I could see the cursor and pointed it to menu item and clicked, there was no response. However I can see the item was 'pressed'.

Ahhh okay, yeah this is sadly the only issue that Avalonia still has on the Steam Deck. We know about it, but aren't sure what's causing it or how to fix it. We are tracking this issue here: #4073

Thank you for taking the time to test things and trying to help us with this issue! :D

**Comment by shineofleo** [comment]
OK just tested 1.1.571.

Ava loaded in game mode, and it also loads game. but the mouse clicking still not working. I recorded some video but no idea how to upload here... 

headless works well.

I added standard Ryujinx as non-steam game, it runs well in game mode. The game has no more black screen.
However, if put it through steam room manager, show the menu, then load the game, black screen occurs. same issue.
Now I have problem loading game via shortcut in SRM it shows 0004 error not finding application but my path should be correct. Not sure what I did wrong but this can be ignored...

**Comment by shineofleo** [comment]
> Ahhh okay, yeah this is sadly the only issue that Avalonia still has on the Steam Deck. We know about it, but aren't sure what's causing it or how to fix it. We are tracking this issue here: #4073
> 
> Thank you for taking the time to test things and trying to help us with this issue! :D

OK maybe I can reply in #4073. Just a quick thinking: ava works fine in desktop mode, but not game mode. Hmmm perhaps there is really something special in gaming mode block both the drop down menu and produce the black screen...

I am glad to help! After all YOU are the heroes! Thanks!

**Comment by wanabrar** [comment]
I rolled back to firmware 3.3.3 or 3.3.2 on the Deck, and it's working fine again on gaming mode. Guess I won't update further till this is actually resolved.

**Comment by thuutien** [comment]
> I rolled back to firmware 3.3.3 or 3.3.2 on the Deck, and it's working fine again on gaming mode. Guess I won't update further till this is actually resolved. I think the problem started on firmware update 3.4.2.

How can you do that?

**Comment by wanabrar** [comment]
> How can you do that?

> It's very simple to roll back: turn off the steam deck, then press Power Button + Three Points Button until you hear the boot sound, then release immediately the power button but keeping press Three Point Button.

If you done right, you will see a little menu with the first voice called "current" and the second one called "previous". Choose the second voice from top and press A.

You will see a lot of strings appears on the screen, it's normal. At some point, the steam deck will reboot and you can see it rollbacks to 3.3.3 and Ryujinx don't have any problem in this version.

The entire procedure will take max 5 minutes in total.

**Comment by TSRBerry** [comment]
> Guess I won't update further till this is actually resolved.
@wanabrar As I previously said, this is most likely a Valve issue, there is not a lot we can do.
Besides Avalonia works fine, so as I also previously said, please use Avalonia.

**Comment by thuutien** [comment]
this is from Emudeck discord. Instructions to update to AVA version for emudeck users.
https://gist.github.com/rawdatafeel/5a64d84b3d112b126649a2416f4e4c1c

**Comment by TSRBerry** [comment]
This script is not a good workaround. It would be better to change the Steam shortcuts accordingly and not fiddle with the files.

Ryujinx has an auto-updater and it would break after using this script.

**Comment by Eolf1337** [comment]
> this is from Emudeck discord. Instructions to update to AVA version for emudeck users. https://gist.github.com/rawdatafeel/5a64d84b3d112b126649a2416f4e4c1c

Thanks this worked great for me

**Comment by shineofleo** [comment]
Ava is the better one thanks for that! Looking forward to the formal release! Usable now anyway :D

**Comment by Eolf1337** [comment]
> This script is not a good workaround. It would be better to change the Steam shortcuts accordingly and not fiddle with the files.
> 
> Ryujinx has an auto-updater and it would break after using this script.

It still kinda works. Hours after running the script i got an alert to update from Ryujinx, update and re-applied the script, everything works.

**Comment by TSRBerry** [comment]
Yeah, but that's a hassle and not the point of auto updating. You should be able to just run Ryujinx normally after applying an update.

This script breaks that functionality, so I'd like to advise changing the steam shortcuts instead of renaming the executable every update.

**Comment by Her0es** [comment]
> this is from Emudeck discord. Instructions to update to AVA version for emudeck users.
> https://gist.github.com/rawdatafeel/5a64d84b3d112b126649a2416f4e4c1c

THANK YOU FOR THIS! This worked! Also, Ava build is pretty snazzy. I like.

**Comment by nekoworkshop** [comment]
For those looking for a cleaner workaround, you can grab the proper AVA version at https://github.com/Ryujinx/release-channel-master/releases then extract it. No need to use a script to mess with an existing non-AVA installation. Just point your Steam game path to the extracted ava version and you're set.

**Comment by 0xDEADCADE** [comment]
[A commenter on the issue above](https://github.com/Plagman/gamescope/issues/773#issuecomment-1410483339) mentioned turning on "Force Composition" in the SteamOS Developer settings appears to resolve the problem. Any potential battery life/performance impacts are unknown to me at this time.

**Comment by Inphamiculus** [comment]
I had the exact same issue on my deck across all recent versions of SteamOS. Found out going into Ryujinx settings and changing graphics resolution from native to 2x fixed my issues.

**Comment by Ducolamia** [comment]
> [A commenter on the issue above](https://github.com/Plagman/gamescope/issues/773#issuecomment-1410483339) mentioned turning on "Force Composition" in the SteamOS Developer settings appears to resolve the problem. Any potential battery life/performance impacts are unknown to me at this time.

Had the same issue. This fixed it for me as well.

**Comment by TSRBerry** [comment]
The real issue will be fixed in a future SteamOS update.

**Comment by gustakasn0v** [comment]
> [A commenter on the issue above](https://github.com/Plagman/gamescope/issues/773#issuecomment-1410483339) mentioned turning on "Force Composition" in the SteamOS Developer settings appears to resolve the problem. Any potential battery life/performance impacts are unknown to me at this time.

The setting is called “Force Composite”, and seems to work. I’m using Ryujinx 1.1.621, Steam Deck, SteamOS 3.4.4 stable channel.

Is anybody on this thread knowledgeable about the impact this could have on performance? I’m asking for my Steam Deck, but even a general answer could work.

**Comment by Ducolamia** [comment]
> > [A commenter on the issue above](https://github.com/Plagman/gamescope/issues/773#issuecomment-1410483339) mentioned turning on "Force Composition" in the SteamOS Developer settings appears to resolve the problem. Any potential battery life/performance impacts are unknown to me at this time.
> 
> The setting is called “Force Composite”, and seems to work. I’m using Ryujinx 1.1.621, Steam Deck, SteamOS 3.4.4 stable channel.
> 
> Is anybody on this thread knowledgeable about the impact this could have on performance? I’m asking for my Steam Deck, but even a general answer could work.

I can confirm I have not seen any performance dip with any games when testing on Ryujinx with this enabled. There was virtually no difference between using the workaround with the Avalonia build and the "Force Composite" quick fix.
As mentioned by the devs the real issue should be solved with a Steam OS update but for now there hasn't been any issues.

**Comment by Albert1708** [comment]
> I had the exact same issue on my deck across all recent versions of SteamOS. Found out going into Ryujinx settings and changing graphics resolution from native to 2x fixed my issues.

Graphics resolution from native to 2x did not work for me but in the steamOS if you go to the game properties and set the to lowest resolution then the black screen gets fixed. So my guess the steam OS cant scale the resolution properly yet.

**Comment by TSRBerry** [comment]
As @piplup55 reported, due to a recent [SteamOS update](https://store.steampowered.com/news/app/1675200/view/3675538825744699879) this issue can now finally be closed!
